### DISCUSSION TITLE
umimax运行单元测试报错

### DISCUSSION THREAD

**Discussion by Bclound** [umijs/umi#11619] [Q&A]
![image](https://github.com/umijs/umi/assets/57136514/8f0b05c8-589a-4389-aa10-4f8cb227365f)
目前项目开启了国际化，运行单元测试报错，有解吗？

**Comment by fz6m** [comment]
给一个最小复现看看。

**Reply by Bclound** [reply]
![image](https://github.com/umijs/umi/assets/57136514/f47a815f-5af7-4c54-951b-3e8ddd0c0c96)
经过测试是因为const umi = jest.requireActual('@umijs/max');  删了这个就可以了，另外有没有办法解决跑单元测试报错TypeError: (0 , import_max.matchRoutes) is not a function这种类似的错误，下面那些模拟都是为了解决类似问题的
@fz6m

**Reply by fz6m** [reply]
当涉及到需要 umi 上下文环境和路由信息的组件级测试时，可以考虑使用 `<TestBrowser />` ，一个 case 见 https://github.com/ant-design/ant-design-pro/blob/ccb7a8e69c5baa92158c987b8cc1d1707a791e4f/src/pages/User/Login/login.test.tsx#L32
